src/transfer/gcs.py

```diff
@@ -0,0 +1,122 @@
+import logging
+from typing import List, Union
+
+import google.auth
+import google.auth.exceptions
+from google.cloud import storage
+from google.cloud.storage import Client
+from util.fileutils import make_cloud_paths, collect_filepaths
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+def create_client() -> Client:
+    try:
+        logger.info("Retrieving GCS credentials")
+        credentials, project = google.auth.default()
+        logger.info("Creating client")
+        client = storage.Client(
+            credentials=credentials,
+            project=project,
+        )
+    except Exception as e:
+        logger.error(
+            f"Error while authenticating Google client. Verify json file. \n{e}"
+        )
+        raise
+    return client
+
+
+class GCSUploader:
```

Comment:
Remind me again - we wrapped the google storage APIs directly instead of `cloudfiles` for performance reasons - is that right?

Reply:
In my first small scale test `cloudfiles` performed roughly equivalently to `gsutil` on a single 4-core machine ~500MB/s. The reason I didn't move forward with `cloudfiles` was due to the requirement of having to load each file into memory as a binary string in order to upload it as opposed to a stream. See https://github.com/seung-lab/cloud-files/blob/940850618209b474e46d69e319a0b15b8e7ff194/cloudfiles/cloudfiles.py#L459. However, given the potential performance improvement relative to the regular Python storage API this could be worth re-visiting. The `transfer_to` method can accept a list of paths, so it wouldn't be much effort to swap it out https://github.com/seung-lab/cloud-files/blob/940850618209b474e46d69e319a0b15b8e7ff194/cloudfiles/cloudfiles.py#L874